src/components/SelectionList/TransactionListItemRow.tsx

```diff
@@ -0,0 +1,225 @@
+import React from 'react';
+import type {StyleProp, ViewStyle} from 'react-native';
+import {View} from 'react-native';
+import type {OnyxEntry} from 'react-native-onyx';
+import Button from '@components/Button';
+import Icon from '@components/Icon';
+import * as Expensicons from '@components/Icon/Expensicons';
+import {PressableWithFeedback} from '@components/Pressable';
+import TextWithTooltip from '@components/TextWithTooltip';
+import useLocalize from '@hooks/useLocalize';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import * as CurrencyUtils from '@libs/CurrencyUtils';
+import * as TransactionUtils from '@libs/TransactionUtils';
+import CONST from '@src/CONST';
+import type {Transaction} from '@src/types/onyx';
+import type {SearchTransactionType} from '@src/types/onyx/SearchResults';
+import ExpenseItemHeader from './ExpenseItemHeader';
+import TextWithIconCell from './TextWithIconCell';
+import type {TransactionListItemType} from './types';
+import UserInfoCell from './UserInfoCell';
+
+type TransactionListItemRowProps = {
+    item: TransactionListItemType;
+    showTooltip: boolean;
+    isDisabled: boolean;
+    canSelectMultiple: boolean;
+    onButtonPress: () => void;
+    showItemHeaderOnNarrowLayout?: boolean;
+    containerStyle?: StyleProp<ViewStyle>;
+};
+
+const getTypeIcon = (type?: SearchTransactionType) => {
+    switch (type) {
+        case CONST.SEARCH_TRANSACTION_TYPE.CASH:
+            return Expensicons.Cash;
+        case CONST.SEARCH_TRANSACTION_TYPE.CARD:
+            return Expensicons.CreditCard;
+        case CONST.SEARCH_TRANSACTION_TYPE.DISTANCE:
+            return Expensicons.Car;
+        default:
+            return Expensicons.Cash;
+    }
+};
+
+function TransactionListItemRow({item, showTooltip, isDisabled, canSelectMultiple, onButtonPress, showItemHeaderOnNarrowLayout = true, containerStyle}: TransactionListItemRowProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const theme = useTheme();
+    const {isLargeScreenWidth} = useWindowDimensions();
+    const StyleUtils = useStyleUtils();
+
+    function getMerchant() {
+        const merchant = TransactionUtils.getMerchant(item as OnyxEntry<Transaction>);
+        return merchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT || merchant === CONST.TRANSACTION.DEFAULT_MERCHANT ? '' : merchant;
+    }
```

Comment:
_// I know it looked like this already, but I never noticed_ 🙃 

this function does not have to be defined within the component body, move it outside.

Also the check is a bit confusing, we test 2 values and if they are true then we return nothing, otherwise we return `merchant`.

Is it possible to reverse the logical expression so that it will work like this: `[...] ? merchant : ''`?